src/core/Mixer.cpp

```diff
@@ -292,7 +297,7 @@ bool Mixer::criticalXRuns() const
 
 void Mixer::pushInputFrames( sampleFrame * _ab, const f_cnt_t _frames )
 {
-	lockInputFrames();
+	requestChangeInModel();
```

Comment:
(redirecting the conversation from Gitter)

So, @jasp00 could you explain how you're removing locks here?
It looks very interesting, but I can't quite get what you're replacing the locks with.